File: 1154-day-of-the-year/1154-day-of-the-year.py
class Solution:
    def dayOfYear(self, date: str) -> int:
        y,m,d= map(int, [date[:4], date[5:7], date[8:]])
        days= [0, 31, 59, 90, 120, 151, 181, 212, 243, 273, 304, 334]
        dateval= d+days[m-1]
        if m>2 and (y%4==0 and y%100!=0 or y%400==0):
            dateval= dateval+1
        return dateval

File: 1154-day-of-the-year/1154-day-of-the-year.cpp
class Solution {
public:
    int dayOfYear(string date) {
        int y= stoi(date.substr(0,4));
        int m= stoi(date.substr(5,2));
        int d= stoi(date.substr(8,2));
        vector<int> days= {0, 31, 59, 90, 120, 151, 181, 212, 243, 273, 304, 334};
        int findate= d+days[m-1];
        if (m>2 && ((y%4==0&& y%100!=0) || (y%400==0) )){
            findate++;
        }
        return findate;
    }
};